qos/evm/endpoint.go

```diff
@@ -0,0 +1,48 @@
+package evm
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// endpoint captures the details required to validate an EVM endpoint.
+type endpoint struct {
+	ChainID string
```

Comment:
Why public?

Reply:
Signalling that other methods in the package may need to modify this directly.
This is mostly a preference rather than a strong opinion though, because `endpoint` has package scope.